preprocessing/preprocessing_one_hot_encoding.py

```diff
@@ -59,74 +62,129 @@ def get_preprocessed_brfss_dataset_one_hot_encoded_all_columns(target_one_hot_en
+        data_train, target_train = oversample_dataset(data_train, target_train)
+        if target_one_hot_encoded:
+            target_train = one_hot_encode_target(target_train)
+            target_validation = one_hot_encode_target(target_validation)
+        return data_train, data_validation, target_train, target_validation
+
+
+# Returns a train/validation split of the preprocessed dataset
 # where all columns with ordinal values are one hot encoded (including yes/no columns).
-# If the parameter is set to true, the target column will also be one hot encoded.
+# If the parameter target_one_hot_encoded is set to true, the target column will also be one hot encoded.
+# If the parameter include_test_data is set to true, test data will also be returned.
 # This function includes undersampling
 def get_preprocessed_brfss_dataset_one_hot_encoded_all_columns_train_test_split_undersampled(
-        target_one_hot_encoded=False) -> Tuple[DataFrame, DataFrame, DataFrame, DataFrame]:
-    data_train, data_test, target_train, target_test = \
-        get_preprocessed_brfss_dataset_one_hot_encoded_all_columns_train_test_split(False)
-    data_train, target_train = undersample_dataset(data_train, target_train)
-    if target_one_hot_encoded:
-        target_train = one_hot_encode_target(target_train)
-        target_test = one_hot_encode_target(target_test)
-    return data_train, data_test, target_train, target_test
+        target_one_hot_encoded=False, include_test_data=False) -> Union[
+    Tuple[DataFrame, DataFrame, DataFrame, DataFrame], Tuple[
+        DataFrame, DataFrame, DataFrame, DataFrame, DataFrame, DataFrame]]:
+    if include_test_data:
+        data_train, data_validation, data_test, target_train, target_validation, target_test = \
+            get_preprocessed_brfss_dataset_one_hot_encoded_all_columns_train_test_split(False, include_test_data)
+        data_train, target_train = undersample_dataset(data_train, target_train)
+        if target_one_hot_encoded:
+            target_train = one_hot_encode_target(target_train)
+            target_validation = one_hot_encode_target(target_validation)
+            target_test = one_hot_encode_target(target_test)
+        return data_train, data_validation, data_test, target_train, target_validation, target_test
+    else:
+        data_train, data_validation, target_train, target_validation = \
+            get_preprocessed_brfss_dataset_one_hot_encoded_all_columns_train_test_split(False, include_test_data)
+        data_train, target_train = undersample_dataset(data_train, target_train)
+        if target_one_hot_encoded:
+            target_train = one_hot_encode_target(target_train)
+            target_validation = one_hot_encode_target(target_validation)
+        return data_train, data_validation, target_train, target_validation
 
 
 def one_hot_encode_target(target) -> DataFrame:
```

Comment:
Just as a side note, I think we should change this with a further PR because it is not the best approach. It works, but I think ideally we should fit the encoder once and then just transform the validation and test target.